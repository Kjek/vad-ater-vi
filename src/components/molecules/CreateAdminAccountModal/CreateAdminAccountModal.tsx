import InputButton from '@component/atoms/Button/Button';
import TextBox from '@component/atoms/TextBox/TextBox';
import { api } from '@util/api';
import { toastError, toastSuccessful } from '@util/toast-utils';
import { useRef } from 'react';
import type { Id } from 'react-toastify';
import { toast } from 'react-toastify';

export interface CreateAdminAccountModalProps {
  toggleModal: () => void;
}

const CreateAdminAccountModal = ({
  toggleModal,
}: CreateAdminAccountModalProps) => {
  const toastId = useRef<Id>(0);
  const username = useRef<string>();
  const password = useRef<string>();
  const secret = useRef<string>();
  const { mutate: createAdminAccount } =
    api.admin.createAdminAccount.useMutation({
      onSuccess() {
        toastSuccessful(toastId.current);
        setTimeout(toggleModal, 2000);
      },
      onError(error) {
        toastError(toastId.current, error);
      },
      onMutate() {
        toastId.current = toast.loading('Updating in progress...');
      },
    });

  const handleOnClick = () => {
    if (username.current && password.current && secret.current) {
      createAdminAccount({
        username: username.current,
        password: password.current,
        secret: secret.current,
      });
    }
  };

  return (
    <div className='absolute right-0 top-0 h-full w-full backdrop-blur-sm'>
      <div className='flex h-full w-full items-center justify-center'>
        <div className='dark:bg-gray-custom flex max-w-6xl flex-col flex-wrap gap-4 bg-white p-6'>
          <TextBox
            placeholder='Username'
            onChange={(event) => (username.current = event.target.value)}
          />
          <TextBox
            placeholder='Password'
            variant='password'
            onChange={(event) => (password.current = event.target.value)}
          />
          <TextBox
            placeholder='Secret'
            variant='password'
            onChange={(event) => (secret.current = event.target.value)}
          />
          <InputButton
            value='Generate'
            aria-label='Generate'
            onClick={handleOnClick}
          />
        </div>
      </div>
    </div>
  );
};

export default CreateAdminAccountModal;
