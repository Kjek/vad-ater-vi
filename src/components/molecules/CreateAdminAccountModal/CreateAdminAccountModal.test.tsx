import { fireEvent, render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import CreateAdminAccountModal from './CreateAdminAccountModal';

vi.mock('@util/api', () => ({
  api: {
    admin: {
      createAdminAccount: {
        useMutation: () => ({
          mutate: vi.fn(),
        }),
      },
    },
  },
}));

describe(CreateAdminAccountModal, () => {
  it('renders correctly', () => {
    const toggleModal = vi.fn();
    const { container } = render(
      <CreateAdminAccountModal toggleModal={toggleModal} />
    );
    expect(container).toBeInTheDocument();
    expect(container).toBeVisible();
  });

  it('matches snapshot', () => {
    const defaultRender = render(
      <CreateAdminAccountModal toggleModal={vi.fn()} />
    );
    expect(defaultRender).toMatchSnapshot();
  });
});
