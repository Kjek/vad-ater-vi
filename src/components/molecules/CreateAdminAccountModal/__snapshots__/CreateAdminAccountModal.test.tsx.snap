// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CreateAdminAccountModal > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="absolute right-0 top-0 h-full w-full backdrop-blur-sm"
      >
        <div
          class="flex h-full w-full items-center justify-center"
        >
          <div
            class="dark:bg-gray-custom flex max-w-6xl flex-col flex-wrap gap-4 bg-white p-6"
          >
            <input
              class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
              placeholder="Username"
              type="text"
            />
            <input
              class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
              placeholder="Password"
              type="password"
            />
            <input
              class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
              placeholder="Secret"
              type="password"
            />
            <input
              aria-label="Generate"
              class="cursor-pointer rounded-lg border border-gray-300 px-5 py-2.5 text-sm font-medium text-gray-800 hover:bg-gray-200 dark:border-gray-600 dark:text-gray-300 dark:hover:bg-gray-700 dark:hover:text-white"
              type="button"
              value="Generate"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="absolute right-0 top-0 h-full w-full backdrop-blur-sm"
    >
      <div
        class="flex h-full w-full items-center justify-center"
      >
        <div
          class="dark:bg-gray-custom flex max-w-6xl flex-col flex-wrap gap-4 bg-white p-6"
        >
          <input
            class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
            placeholder="Username"
            type="text"
          />
          <input
            class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
            placeholder="Password"
            type="password"
          />
          <input
            class="w-full self-center rounded-lg border border-gray-300 bg-gray-50 p-2.5 text-sm text-gray-800 outline-none focus:border-gray-300 focus:ring-gray-300 dark:border-gray-600 dark:bg-gray-800 dark:text-gray-300 dark:placeholder-gray-400 dark:focus:border-gray-600 dark:focus:ring-gray-600"
            placeholder="Secret"
            type="password"
          />
          <input
            aria-label="Generate"
            class="cursor-pointer rounded-lg border border-gray-300 px-5 py-2.5 text-sm font-medium text-gray-800 hover:bg-gray-200 dark:border-gray-600 dark:text-gray-300 dark:hover:bg-gray-700 dark:hover:text-white"
            type="button"
            value="Generate"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
